src/Database/LSMTree/Internal.hs

```diff
@@ -526,11 +553,24 @@ data Table m h = Table {
     , tableState        :: !(RWVar m (TableState m h))
     , tableArenaManager :: !(ArenaManager (PrimState m))
     , tableTracer       :: !(Tracer m TableTrace)
+      -- | Session-unique identifier for this table.
+      --
+      -- INVARIANT: a table's identifier is never changed during the lifetime of
+      -- the table.
+    , tableId           :: !Word64
```

Comment:
Technically yes, because it's a record type. There's nothing stopping us from internally constructing a `Table m h -> Table m h` function that changes `tableId`. It's a silly example, I admit :P I thought it would be at least useful to mention that we shouldn't change `tableId` at any point